2324-wt/abstract-interpretation/interim/poster.tex

```diff
@@ -0,0 +1,221 @@
+	\end{tikzpicture}\par
+	% \paragraph{Step 3} The last step uses the decorated dataflow graph to determine wich
+	% path can never be executed. The identified path will then be discarded from the
+	% graph.\par
+	\paragraph{Step 3} The last step uses the decorated dataflow graph to determine and
+	track different properties. We can, e.g., catch endless loops, or, as shown above,
+	eliminate dead paths in the dataflow graph.\par
+\end{multicols}\bigskip
+\endgroup
+
+\begin{multicols}{3}
+	\begin{minipage}{\dimexpr2\columnwidth+\columnsep\relax}
+		\section*{How abstract interpretation fits into \flowr}
+		\paragraph{\flowr}\flowr{} is a static \emph{dataflow analyzer} and
+		\emph{program slicer} for the R language. Program slicers reduce a given program
+		to a set of statements --- the so called slices --- that influence a variable at
+		a specific point in the program --- the slicing criterion. \flowr{} formulates
+		the slicing as a reachability problem on the dataflow graph, which itself is
+		based on the abstract syntax tree of the given R~program.
+		\paragraph{Abstract Interpretation} Abstract interpretation is all about
+		figuring out \emph{runtime properties} of a given program without actually
+		executing it. These properties include aspects like the sign or nullness of a
+		variable. For now we decided to limit ourselves to determining the \emph{domain
+		of numeric variables}. In other words, we want to know what values a variable
+		can have at a certain point in the program.
+		\paragraph{Abstract interpretation in \flowr} Abstract interpretation helps
+		\flowr{} to further \emph{narrow down the slice} of a program. This can be done
+		by removing certain paths from the dataflow graph if we can prove that they will
+		never be executed. Imagine a conditional like \rc{if(i < 10) doit() else run()}.
+		If, through abstract interpretation, we know that \rc{i} is always smaller than
+		\rc{10}, there's no need to include the path stemming from the \rc{if}'s
+		else-case in the slice. % TODO: is this okay? Or does it read like we only catch dead code
+		\section*{Abstract Interpretation}
+		\begin{tikzpicture}
+			\node[roundednode,minimum width=\textwidth,minimum height=0.2\textheight,draw=black,fill=lightgray!20] {\Large Maybe an example?!};
+		\end{tikzpicture}
+		\lipsum[1-2]
+	\end{minipage}\vfill\columnbreak\null\columnbreak
+	\section*{Project Organization}
+	\paragraph{Organization} We're doing \emph{weekly meetings}, where we discuss the
+	progress made in the past week, talk about open issues, and prioritize tasks for the
+	next week. We also keep a \emph{record of important things} that came up in our
+	meeting. If there's any spare time, Florian often tells me about new
+	\sout{weird}cool things he learned about the R language.
+	\paragraph{Tech Stack} \flowr{} is developed with \emph{TypeScript}, then compiled
+	down to JavaScript with Node.js as its runtime. While we use different libraries for
```

Comment:
compiled down suggeriert way too much work :P

Reply:
Fixed in ad36492